src/components/screens/consul/Proposals/NewProposal.js

```diff
@@ -117,6 +117,28 @@ export const NewProposal = ({ navigation, data, query }) => {
     setValue('tagList', filterData.toString());
   }, [tags]);
 
+  const uploadImage = async (imageData) => {
+    let imageAttributes;
+    const imageUploadData = await uploadAttachment(imageData, 'image');
+
+    if (imageUploadData.status === 200) {
+      const { cached_attachment: cachedAttachment, filename: title } = JSON.parse(
+        imageUploadData.body
+      );
+
+      imageAttributes = {
+        title,
+        cachedAttachment
+      };
+    } else if (imageUploadData.status === 422) {
+      const errors = JSON.parse(imageUploadData.body).errors.toLowerCase().split(' ').join('-');
+
+      throw errors;
+    }
+
+    return imageAttributes;
+  };
```

Comment:
not bad, thanks. i like the approach.
do we still need `let imageAttributes;` now?

if status is 200 we could

```
return {
  title,
  cachedAttachment
}
```

and then have another if status is 422 to throw an error.

this should be the same but less and more straight forward code i guess? what do you think?